### PR TITLE
Isolate bft getByBlockNumber

### LINKED ISSUE #5162: BFT sequence number usage
Part of #4789

### PR DESCRIPTION
**Pull request by siladu**
Fixes #5162 and #5164 

Introduce BftProtocolSchedule with a copy of MutableProtocolSchedule.getByBlockNumber
This method will be deleted from the ProtocolSchedule interfaces and so will only exist in BftProtocolSchedule (although it may be renamed in a subsequent commit).

Rename existing *BftProtocolSchedule classes to append Builder for clarity (and inline with other ProtocolScheduleBuilder classes)

### CODE REVIEW

File: consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/BftProtocolSchedule.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.hyperledger.besu.ethereum.mainnet.ScheduledProtocolSpec;
+
+public class BftProtocolSchedule extends MutableProtocolSchedule {
+
+  public BftProtocolSchedule(final MutableProtocolSchedule mutableProtocolSchedule) {
+    super(mutableProtocolSchedule);
+  }
+
+  @Override
+  public ProtocolSpec getByBlockNumber(final long number) {
```
Comment (siladu) [inline consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/BftProtocolSchedule.java:31]: Should we rename this or does it still make sense?

File: consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/BaseBftProtocolScheduleBuilder.java [hunk 2]
```diff
@@ -72,15 +72,18 @@ public ProtocolSchedule createProtocolSchedule(
...
-            config,
-            DEFAULT_CHAIN_ID,
-            specAdapters,
-            privacyParameters,
-            isRevertReasonEnabled,
-            config.isQuorum(),
-            evmConfiguration)
-        .createProtocolSchedule();
+    final MutableProtocolSchedule mutableProtocolSchedule =
+        (MutableProtocolSchedule)
```
Comment (jframe) [inline consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/BaseBftProtocolScheduleBuilder.java:76]: Bit nasty need to do this casting, but don't see any way around it

File: consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/BftProtocolSchedule.java (hunk 1 above)
Comment (jframe) [reply consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/BftProtocolSchedule.java:46]: Still make sense, don't think we need to rename it

**Review by jframe — APPROVED** [review]

**Review by jframe — APPROVED** [review]
LGTM

**Pull request merged**